### PR TITLE
Reverted decidability procedures in `Data.Nat.Binary.Properties` to improve performance

### PR DESCRIPTION
**Pull request by MatthewDaggitt**

### CODE REVIEW

File: src/Data/Nat/Binary/Properties.agda [hunk 1]
```diff
@@ -273,11 +274,35 @@ private
...
+... | tri> _ _ y<x =  tri> (>⇒≮ gt) (>⇒≢ gt) gt  where gt = odd<even (inj₁ y<x)
+<-cmp 1+[2 _ ] zero =  tri> (>⇒≮ gt) (>⇒≢ gt) gt  where gt = 0<odd
+<-cmp 1+[2 x ] 2[1+ y ]  with <-cmp x y
+... | tri< x<y _ _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)  where lt = odd<even (inj₁ x<y)
+... | tri≈ _ x≡y _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)  where lt = odd<even (inj₂ x≡y)
+... | tri> _ _ x>y =  tri> (>⇒≮ gt) (>⇒≢ gt) gt  where gt = even<odd x>y
+<-cmp 1+[2 x ] 1+[2 y ]  with <-cmp x y
+... | tri< x<y _  _   =  tri< lt (<⇒≢ lt) (<⇒≯ lt)  where lt = odd<odd x<y
+... | tri≈ _ refl _   =  tri≈ (≡⇒≮ refl) refl (≡⇒≯ refl)
+... | tri> _ _    x>y =  tri> (>⇒≮ gt) (>⇒≢ gt) gt  where gt = odd<odd x>y
```
Comment (gallais) [inline src/Data/Nat/Binary/Properties.agda]: Feels like we should be able to play @laMudri's games on `Tri`
just like we have on `Dec`.

Comment (MatthewDaggitt) [reply src/Data/Nat/Binary/Properties.agda]: Yup, probably true, although won't we need need a three valued logic? Might be best to open an issue for `v1.3`...

Comment (laMudri) [reply src/Data/Nat/Binary/Properties.agda]: At least in this case, we have a clear intended semantics for our three-valued logic (if we choose to see it as such). But yeah, the same pattern clearly applies.

Comment (gallais) [reply src/Data/Nat/Binary/Properties.agda]: We can use the classic `LT`, `EQ`, `GT` familiar to Haskellers.

**Pull request merged**